lib/access-context.js

```diff
@@ -143,19 +144,9 @@ AccessContext.prototype.getUserId = function() {
     // or permit to resolve a valid user model
     var userModel = this.registry.findModel(p.type);
     if (!userModel) continue;
-    if (userModel.modelName === 'User') {
+    if (userModel.prototype instanceof BaseUser) {
       return p.id;
     }
```

Comment:
arff..... yes! :-) i just couldn't see this ! (confusing getModel() with getModelByType() !!)

Then it should do, thanks.

Just one last check: you're confident in model prototypal inheritance in loopback and there's no way a model extending 'User' can return false to `instanceof BaseUser` ?